tlsn/examples/interactive/interactive.rs

```diff
@@ -0,0 +1,204 @@
+        .await
+        .unwrap();
+    let (mpc_tls_connection, prover_fut) =
+        prover.connect(tls_client_socket.compat()).await.unwrap();
+    let mpc_tls_connection = TokioIo::new(mpc_tls_connection.compat());
+    let prover_task = tokio::spawn(prover_fut);
+
+    // MPC-TLS Handshake
+    let (mut request_sender, connection) =
+        hyper::client::conn::http1::handshake(mpc_tls_connection)
+            .await
+            .unwrap();
+
+    let connection_task = tokio::spawn(connection.without_shutdown());
+
+    // MPC-TLS: Send Request and wait for Response
+    let request = Request::builder()
+        .uri(uri.clone())
+        .header("Host", server_domain)
+        .header("Connection", "close")
+        .header("Secret", SECRET)
+        .method("GET")
+        .body(Empty::<Bytes>::new())
+        .unwrap();
+    let response = request_sender.send_request(request).await.unwrap();
+
+    assert!(response.status() == StatusCode::OK);
+
+    // Close TLS Connection
+    let tls_connection = connection_task.await.unwrap().unwrap().io.into_inner();
+    tls_connection.compat().close().await.unwrap();
+
+    // Create proof for the Verifier
+    let prover = prover_task.await.unwrap().unwrap().start_prove();
+    let prover = react_and_reveal_received_data(prover);
+    let mut prover = react_and_reveal_sent_data(prover);
+    prover.prove().await.unwrap();
+
+    // Finalize
+    prover.finalize().await.unwrap()
+}
+
+#[instrument(skip(socket))]
+async fn verifier<T: AsyncWrite + AsyncRead + Send + Sync + Unpin + 'static>(
+    socket: T,
+    id: &str,
+) -> (RedactedTranscript, RedactedTranscript, SessionInfo) {
+    // Setup Verifier
+    let verifier_config = VerifierConfig::builder().id(id).build().unwrap();
+    let verifier = Verifier::new(verifier_config);
+
+    // Verify MPC-TLS and wait for (redacted) data
+    let (sent, received, session_info) = verifier.verify(socket.compat()).await.unwrap();
+
+    // Check send data: check host
+    let sent_data = String::from_utf8(sent.data().to_vec()).expect("Verifier expected sent data");
+    sent_data
+        .find(SERVER_DOMAIN)
+        .unwrap_or_else(|| panic!("Verification failed: Expected host {}", SERVER_DOMAIN));
+    // Check received data: check json and version number
+    let response = parse_response(received.data()).unwrap();
+    let body = response
+        .body
+        .map_or(String::new(), |body| {
+            String::from_utf8_lossy(body.as_bytes()).to_string()
+        })
+        .replace('\0', "");
+    let json =
+        serde_json::from_str::<InfoResponse>(body.as_str()).expect("Expected valid JSON response");
+    json.public_key
+        .find("BEGIN PUBLIC KEY")
+        .expect("Expected valid public key in JSON response");
+    // Check Session info: server name
+    assert_eq!(session_info.server_name.as_str(), SERVER_DOMAIN);
+
+    (sent, received, session_info)
+}
+
+/// Redact and reveal received data to the verifier
```

Comment:
acc to the style guide method comments should be formatted as:
```suggestion
/// Redacts and reveals received data to the verifier.
```